### PR TITLE
fix(scheduler): scope required filters to delivered tabs

### LINKED ISSUE #27333: [Scheduled deliveries] can't create a delivery for a single tab unless required filters from other tabs are also set
## Description

When creating a scheduled delivery for a **specific tab** of a dashboard, the "Create schedule" button stays disabled unless **all** required filters on the dashboard have values — including required filters that only apply to charts on *other* tabs that aren't part of the delivery.

The required-filters check for scheduled deliveries appears to validate against the whole dashboard's required filters, instead of only the filters that are required for the tab(s) actually selected in the delivery.

## Setup

A dashboard with two tabs, each with its own required filter scoped to that tab:

- **Tab 1**: required filter on `Order date month` (e.g. August)
- **Tab 2**: required filter on `Email starts with` (e.g. "a")

On the dashboard itself these behave independently — setting only the email filter loads tab 2 fine, even with the order date filter empty. So tab scoping works correctly for viewing.

## Expected behavior

"Create schedule" is enabled — the only tab included in the delivery has its required filter satisfied. Required filters scoped to tabs that aren't part of the delivery shouldn't block it.

## Actual behavior

"Create schedule" stays greyed out until a value is also provided for the `Order date month` filter, which is only required on tab 1 (not included in the delivery). It only becomes enabled once **both** required filters have values. Removing the other tab's filter from the delivery's filter list doesn't help either — the button remains disabled.

## Notes

- The delivery validation likely predates tab-scoped required filters and doesn't account for tab selection.
- Video walkthrough available here: https://www.loom.com/share/20bd4e61168241ce859baf79a0d000a5

## Steps to reproduce

1. Clear the required filter values on the dashboard
2. Click **Deliveries → New delivery**, add a name and recipient email
3. Under **Data & format**, select only tab 2
4. A message shows that all required filters must have a value
5. Set a value for the email filter (the only filter required on the selected tab)

### version

v1.146.5

### Cloud or self-hosting

None

### PR DESCRIPTION
**Pull request by jelstongreen**
Closes: lightdash/lightdash#27333

## Summary

Scheduled deliveries for selected dashboard tabs no longer require values for filters that only affect omitted tabs. Submit, Send once now, and visible warnings now share the same tab-scoped requirement result.

## Root cause

The form evaluated required dimensions and metrics against the whole dashboard. The initial scoping implementation also evaluated grouped requirements before excluding omitted-tab members and treated automatic filters as if they applied to every tile.

```text
Before: selected tabs → whole-dashboard requirements → blocked by omitted tab
After:  selected tabs + tile fields → delivery requirements → validate and warn
```

## Fix

The scheduler now derives applicability from the Dashboard provider-owned tile field map and the authoritative dashboard tab list. It scopes individual and grouped requirements before evaluation, uses normal automatic filter-to-tile matching, and preserves conservative validation while data loads or all tabs are delivered.

* `filterRequirements.ts` scopes requirements to selected tabs before group evaluation.
* `useSchedulerFormModal.ts` produces one scoped result for submit, Send once now, and warnings.
* Scheduler component and hook tests cover partial tabs, explicit all-tabs, empty omitted tabs, automatic targets, loading, and mixed-tab groups.

## Test plan

- [X] `pnpm -F frontend test src/features/scheduler` — 5 files, 42 tests passed
- [X] `pnpm -F frontend test` — 335 files, 2552 tests passed
- [X] `pnpm -F frontend typecheck`
- [X] `pnpm -F frontend lint`
- [X] `pnpm -F frontend format`
- [ ] Manual browser verification — no browser runtime was available in this environment

### CODE REVIEW

File: packages/frontend/src/features/scheduler/utils/filterRequirements.ts [hunk 1]
```diff
@@ -33,7 +81,14 @@ export const getSchedulerFilterRequirements = (
         ),
     };
 
-    const unmetRequirements = getUnmetFilterRequirements(effectiveFilters);
+    const appliesToDelivery = getAppliesToDelivery(tabScope);
+    const unmetRequirements = getUnmetFilterRequirements(
```
Comment (sdolidze) [inline packages/frontend/src/features/scheduler/utils/filterRequirements.ts]: **Blocker:** Group satisfaction is evaluated before excluded-tab members are removed. A populated member on an omitted tab can therefore satisfy the group while the selected tab member remains empty, enabling submission and Send Now incorrectly. Excluded members also leak into the warning list. Please filter the effective dimension and metric rules with `appliesToDelivery` before calling `getUnmetFilterRequirements`, and add a mixed-tab group test where the omitted member is populated and the selected member is empty.

File: packages/frontend/src/features/scheduler/utils/filterRequirements.ts [hunk 2]
```diff
@@ -1,16 +1,63 @@
...
+ * applies everywhere, otherwise it must reach a tile in the delivery. Deciding
+ * that needs each tile's filterable fields, so while those are missing nothing
+ * is scoped out.
+ */
+const getAppliesToDelivery = (
+    tabScope: SchedulerTabScope | undefined,
+): ((filter: DashboardFilterRule) => boolean) => {
+    if (
+        !tabScope ||
+        tabScope.selectedTabs === null ||
```
Comment (sdolidze) [inline packages/frontend/src/features/scheduler/utils/filterRequirements.ts]: The PR promises conservative behavior when all tabs are included, but this only recognizes `selectedTabs === null`. Create mode initializes the all-tabs state as the full array of tab UUIDs, and the UI also treats that array as “Include all dashboard tabs.” In that normal state this branch can scope out requirements and change previous behavior. Please normalize the complete list to `null`, or detect that every dashboard tab is selected, and add a full-array regression test.

File: packages/frontend/src/features/scheduler/hooks/useSchedulerFormModal.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ import {
 import { Limit } from '../components/types';
 import { getSchedulerFilterRequirements } from '../utils/filterRequirements';
 import { useScheduler, useSendNowScheduler } from './useScheduler';
+import { useSchedulerFilterableTiles } from './useSchedulerFilterableTiles';
```
Comment (sdolidze) [inline packages/frontend/src/features/scheduler/hooks/useSchedulerFormModal.ts]: The required frontend formatting gate is currently red. CI reports formatting failures in this file and `filterRequirements.test.ts`. Please run `pnpm -F frontend fix-format` and rerun the checks.

File: packages/frontend/src/features/scheduler/utils/filterRequirements.test.ts [hunk 4]
```diff
@@ -147,3 +154,121 @@ describe('getSchedulerFilterRequirements', () => {
...
+        required: true,
+        target: { fieldId: 'orders_tab2', tableName: 'orders' },
+        tileTargets: {
+            'tile-1': false,
+            'tile-2': { fieldId: 'orders_tab2', tableName: 'orders' },
+        },
+    });
+    const filters = dashboardFilters([tab1Filter, tab2Filter]);
+
+    it('only flags requirements that apply to the selected tabs', () => {
```
Comment (sdolidze) [inline packages/frontend/src/features/scheduler/utils/filterRequirements.test.ts:205]: These tests exercise the pure helper with a hand-built scope, but do not cover the new wiring through `useSchedulerFilterableTiles`, form validation/Send Now, and filter warnings. They would stay green if one of those callers stopped passing `selectedTabs`. Please add a hook/component integration test that selects tab 2, verifies tab 1 does not block or warn, then selects all tabs and verifies it blocks.

**Review by sdolidze — CHANGES_REQUESTED** [review]
Requesting changes for the grouped-requirement scoping bug and the failing frontend formatting gate. I also left inline comments for the all-tabs representation and integration-test coverage.

File: packages/frontend/src/features/scheduler/utils/filterRequirements.ts (hunk 1 above)
Comment (sdolidze) [reply packages/frontend/src/features/scheduler/utils/filterRequirements.ts]: Fixed in c5b9c128f3. Requirements are scoped before grouped evaluation, with a mixed-tab group regression covering the omitted populated member.

File: packages/frontend/src/features/scheduler/hooks/useSchedulerFormModal.ts (hunk 3 above)
Comment (sdolidze) [reply packages/frontend/src/features/scheduler/hooks/useSchedulerFormModal.ts]: Fixed in c5b9c128f3. Frontend format now passes, alongside lint and typecheck.

File: packages/frontend/src/features/scheduler/utils/filterRequirements.ts (hunk 2 above)
Comment (sdolidze) [reply packages/frontend/src/features/scheduler/utils/filterRequirements.ts]: Fixed in c5b9c128f3. Explicit full-tab selections are compared with authoritative dashboard tab UUIDs, including an omitted empty-tab regression.

File: packages/frontend/src/features/scheduler/utils/filterRequirements.test.ts (hunk 4 above)
Comment (sdolidze) [reply packages/frontend/src/features/scheduler/utils/filterRequirements.test.ts:205]: Fixed in c5b9c128f3. Hook and component seam tests now verify selected-tab submission gating and the shared warning result; the full frontend suite passes.

**Review by sdolidze — APPROVED** [review]
Addressed the tab-scoping, grouped-requirement, automatic-filter, all-tabs, formatting, and integration-test findings in c5b9c128f3. Local scheduler tests, the full frontend suite, typecheck, lint, and format all pass.
